e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.Registry.cs

```diff
@@ -196,8 +196,7 @@ public static class Registry {
                     string discoveryUrl,
                     CancellationToken ct = default) {
 
-                var applicationId = await GetApplicationIdAsync(context, discoveryUrl, ct);
-
+                var applicationId = context.ApplicationId == null ? await GetApplicationIdAsync(context, discoveryUrl, ct) : context.ApplicationId;
```

Comment:
Nit: For readability I prefer ternary to not include method calls.  Better a if else here, or first assign applicationId from context, then test for null and retrieve it "async" as "fall back".  A comment would also be helpful.

Reply:
Right, will change that. Ideally the entire cleanup logic needs to be reviewed because is inconsistent.